### ISSUE TITLE
More issues with dataflow in top-level code

### ISSUE THREAD

**Issue by boriscc** [facebook/pyre-check#558]
# Pysa Bug
**Pre-submission checklist**
[X] I've checked the list of [common issues](https://pyre-check.org/docs/pysa-quickstart/#common-issues) and mine does not appear

**Bug description**
Related to bug #550. Using nightly, the example in that bug report now works, i.e. it reports an issue on

```python
import os

b = os.chdir("test")
exec(b)
```

when written in the top level. However, it will still fail to detect an issue if exec is wrapped in a function:

```python
import os

def foobar(d):
    exec(d)

foobar(os.chdir("test"))
a = os.chdir("test")
foobar(a)
```

None of the above two calls to foobar will trigger an issue. However, if I put the top-level code above into another function, it will find the issues

**Reproduction steps**
Run the code snippet above with nightly 0.0.101640088603 with Python 3.8.12 with a rule connecting the source os.chdir with the sink exec.

**Expected behavior**
Both calls to foobar should be reported as issues.

**Logs**
Unfortunately I am not able to post logs right now, if you need them I can provide them at a later time.

**Additional context**
I use Anaconda for my virtual environment

**Comment by arthaud** [comment]
Thanks a lot for the report! This is actually a source of non determinism in the analysis and we have been tracking those for a long time. This is a bug due to not tracking top levels in the call graph and dependency graph. A fix will be out shortly.

**Comment by arthaud** [comment]
Fixed by e2c7b63f871e7aedf4336adae61f5da06ef59ec5.
